src/Surfnet/Stepup/Tests/Configuration/Value/AllowedSecondFactorListTest.php

```diff
@@ -144,6 +144,76 @@ public function a_second_factor_not_on_the_allowed_second_factor_list_is_not_all
         );
     }
 
+    /**
+     * @test
+     * @group domain
+     */
+    public function an_allowed_second_factor_list_equals_an_allowed_second_factor_list_with_the_same_values()
+    {
+        $allowedSecondFactorList = AllowedSecondFactorList::ofTypes([
+            new SecondFactorType('sms'),
+            new SecondFactorType('yubikey'),
+        ]);
+        $sameAllowedSecondFactorList = AllowedSecondFactorList::ofTypes([
+            new SecondFactorType('sms'),
+            new SecondFactorType('yubikey'),
+        ]);
+
+        $isSameAllowedSecondFactorList = $allowedSecondFactorList->equals($sameAllowedSecondFactorList);
+
+        $this->assertTrue(
+            $isSameAllowedSecondFactorList,
+            'An allowed second factor list should equal a second factor list with the same values but it does not'
+        );
+    }
+
+    /**
+     * @test
+     * @group domain
+     */
+    public function an_allowed_second_factor_list_does_not_equal_an_allowed_second_factor_list_with_different_values()
```

Comment:
This, the previous and the next test are a bit... incomplete. They're not covering all cases that might be important. E.g.: if I change the first test (of amount of elements) in the `equals` method to 
```
if (count($other->allowedSecondFactors) < count($this->allowedSecondFactors)) { bork() }
```
all tests still pass. However, the objects won't agree on whether or not they're equal (`$a->equals($b) !== $b->equals($a)`).

They're structurally ok, but not thorough enough. As an example on how we did this in other projects, check https://github.com/OpenConext/SamlValueObject/blob/develop/tests/OpenConext/Value/Saml/EntitySetTest.php